app/cdap/components/RecipeList/RecipeTableRow.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import T from 'i18n-react';
+import { IRecipe } from './types';
+import ActionsPopover, { IAction } from 'components/shared/ActionsPopover';
+import MyDataPrepApi from 'api/dataprep';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import fileDownload from 'js-file-download';
+import ConfirmationModal from 'components/shared/ConfirmationModal';
+import { IState, reset } from './reducer';
+import { format, TYPES } from 'services/DataFormatter';
+
+interface IRecipeTableRowProps {
+  recipe: IRecipe;
+  createdTimeMillis?: any;
+  isShowAllColumns: boolean;
+  isShowActions: boolean;
+  onViewRecipe: (selectedRecipe: IRecipe) => void;
+  onEditRecipe: (selectedRecipe: IRecipe) => void;
+  onSelectRecipe: (selectedRecipe: IRecipe) => void;
+  state: IState;
+  dispatch: (action: any) => void;
+}
+
+const PREFIX = 'features.WranglerNewUI.Recipe';
+const DATE_FORMAT = 'MM-DD-YYYY HH:mm A';
+
+export const RecipeTableRow = ({
+  recipe,
+  isShowAllColumns,
+  isShowActions,
+  onViewRecipe,
+  onEditRecipe,
+  onSelectRecipe,
+  state,
+  dispatch,
+}: IRecipeTableRowProps) => {
+  const [showPopover, setShowPopover] = useState(false);
+  const [isDeleteModalOpen, setDeleteModalOpen] = useState(false);
+  const [isDeleteLoading, setDeleteLoading] = useState(false);
+  const [deleteErrMsg, setDeleteErrMsg] = useState(null);
+  const [extendedDeleteErrMsg, setExtendedDeleteErrMsg] = useState(null);
+
+  const handleActionsClick = (e) => {
+    e.preventDefault();
+  };
+
+  const togglePopover = () => {
+    setShowPopover(!showPopover);
+  };
+
+  const getRecipeDetailsById = (responseHandler) => {
+    MyDataPrepApi.getRecipeById({
```

Comment:
it'd be better if we moved these api calls inside the reducer as well. That way we know where we make all our api calls in case of potential changes. You can import the functions her from the reducer just like you do for `reset`.

Reply:
Yeah. moved all api calls to reducer